library/src/blas2/rocblas_ger.cpp

```diff
@@ -106,51 +88,10 @@ namespace
         if(!m || !n)
             return rocblas_status_success;
 
-        hipStream_t rocblas_stream = handle->rocblas_stream;
-
-        static constexpr int GEMV_DIM_X = 128;
-        static constexpr int GEMV_DIM_Y = 8;
-        rocblas_int          blocksX    = (m - 1) / GEMV_DIM_X + 1;
-        rocblas_int          blocksY    = (n - 1) / GEMV_DIM_Y + 1;
-
-        dim3 ger_grid(blocksX, blocksY);
-        dim3 ger_threads(GEMV_DIM_X, GEMV_DIM_Y);
-
-        if(incx < 0)
-            x -= ptrdiff_t(incx) * (m - 1);
-        if(incy < 0)
-            y -= ptrdiff_t(incy) * (n - 1);
-
-        if(handle->pointer_mode == rocblas_pointer_mode_device)
-            hipLaunchKernelGGL(ger_kernel,
-                               ger_grid,
-                               ger_threads,
-                               0,
-                               rocblas_stream,
-                               m,
-                               n,
-                               alpha,
-                               x,
-                               incx,
-                               y,
-                               incy,
-                               A,
-                               lda);
-        else
-            hipLaunchKernelGGL(ger_kernel,
-                               ger_grid,
-                               ger_threads,
-                               0,
-                               rocblas_stream,
-                               m,
-                               n,
-                               *alpha,
-                               x,
-                               incx,
-                               y,
-                               incy,
-                               A,
-                               lda);
+        rocblas_ger_strided_batched_template(
+            handle, m, n, alpha, x, incx, incx * m, y, incy, incy * n, A, lda, lda * n, 1);
+        // rocblas_ger_template(handle, m, n, alpha, x, incx, y, incy, A, lda);
+
```

Comment:
When calling ger from other routines or libraries, via the template_functions, one may need to work with only a sub-matrix (sub-vector) of A (x or y). In the non-batched case, this can be done passing the pointer(s) shifted the desired amount of positions, but this cannot be done in the batched case, where A, x or y are pointers to pointers.
 
To enable this, batched (and strided_batched for congruency) template_functions should include a "shift" parameter that could be applied to the instances in the batch.  If no shift is required, it should be equal zero... in this case:
rocblas_ger_strided_batched_template(
            handle, m, n, alpha, x, **0**, incx, incx * m, y, **0**, incy, incy * n, A, **0**, lda, lda * n, 1);

*see other related comments.

(Whether to expose these shift parameters to the API could be discussed later)